doc/source/groupby.rst

```diff
@@ -436,51 +419,113 @@ Or for an object grouped on multiple columns:
 Aggregation
 -----------
 
-Once the GroupBy object has been created, several methods are available to
-perform a computation on the grouped data. These operations are similar to the
-:ref:`aggregating API <basics.aggregate>`, :ref:`window functions API <stats.aggregate>`,
-and :ref:`resample API <timeseries.aggregate>`.
+This section describes how to aggregate data. We will be giving examples using
+the ``tips.csv`` dataset. Each row represents a meal at some restaurant; The
+columns store the value of the total bill, the size if the tip and some metadata
+about the customer.
+
+.. ipython:: python
 
-An obvious one is aggregation via the ``aggregate`` or equivalently ``agg`` method:
+   tips = pd.read_csv('./data/tips.csv')
+   tips
+
+What if we wanted to know the average total bill on each day? We split the data
+so that each group consists of all the meals eaten on the same day. We want a
+single value for each group, so we should use the aggregate function:
 
 .. ipython:: python
 
-   grouped = df.groupby('A')
-   grouped.aggregate(np.sum)
+   tips.groupby('day').aggregate('mean')
 
-   grouped = df.groupby(['A', 'B'])
-   grouped.aggregate(np.sum)
+The result has the group names, in this case the days, as the index along the
+grouped axis. Along the other axis we have the columns for which Pandas could
+calculate a mean, i.e. the ones with a numeric data type. We could have selected
+the ``total_bill`` column either before or after aggregating to limit the result
+to this column, but not before splitting since we need the ``day`` column for
+the splitting.
 
-As you can see, the result of the aggregation will have the group names as the
-new index along the grouped axis. In the case of multiple keys, the result is a
-:ref:`MultiIndex <advanced.hierarchical>` by default, though this can be
-changed by using the ``as_index`` option:
+How about the number of guests for each day and for each time of day? In this
+case it is not enough to split the data on the day it was eaten, we also need
+split by the time of day. Instead of calculating the mean, like in the previous
+example we use the ``sum`` function.
 
 .. ipython:: python
 
-   grouped = df.groupby(['A', 'B'], as_index=False)
-   grouped.aggregate(np.sum)
+   tips.groupby(['day', 'time'])['size'].agg('sum')
 
-   df.groupby('A', as_index=False).sum()
+``agg`` is short for ``aggregate``.
 
-Note that you could use the ``reset_index`` DataFrame function to achieve the
-same result as the column names are stored in the resulting ``MultiIndex``:
+Pandas has support for a number of basic descriptive statistic functions which can be used with aggregate:
 
-.. ipython:: python
+.. csv-table::
+    :header: "Function", "Description"
+    :widths: 20, 80
 
-   df.groupby(['A', 'B']).sum().reset_index()
+    ``count``, Number of non-NA observations
```

Comment:
these can be links as well

Reply:
The links to the functions available with aggregate? I think that would be a great idea. Where can I find the list of available functions and the shortcuts? I figured that must have been documented elsewhere at some point but couldnt find it?